sdk/core/System.ClientModel/src/Internal/CancellationTokenExtensions.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ClientModel.Primitives;
+using System.Threading;
+
+namespace System.ClientModel.Internal;
+
+internal static class CancellationTokenExtensions
+{
+    public static RequestOptions? ToRequestOptions(this CancellationToken cancellationToken)
```

Comment:
should we just add it as a static helper to RequestOptions?

Reply:
e.g. `public static RequestOptions FromCancellationToken(CancellationToken cancellationToken)` factory method?  That is doable if you're happy with the addition to the API surface.  If we make RequestContext inherit from RequestOptions in the future we would need to newslot it, would be a consideration?

Reply:
it's a static method, i.e. no newslotting needed or possible. Though, feel free to punt. We can always add this later. I approved the PR

Reply:
Sounds good.  Yes, let's revisit once the Azure.Core 2.0 strategy is pinned down so we can choose an approach that works well with that.

Reply:
Actually, making it an internal factory method on RO is fine.  Addressed.